### DISCUSSION TITLE
Noob troubleshooting sb.click(selector)

### DISCUSSION THREAD

**Discussion by davidjivan** [seleniumbase/SeleniumBase#2817] [Q&A] [answered]
Hi team... can anyone offer ideas how to troubleshoot this?

seleniumbase.common.exceptions.NoSuchElementException: Message: 
 Element {//button[contains(., 'I Accept')]} was not present after 7 seconds!
 
 I am pretty sure I'm using xpath correctly. When I run sb.assert_text('I Accept') it asserts the text. The element is a <button>"I Accept"</button>

Any ideas appreciated!

**Comment by mdmintz** [comment, marked as answer]
If it's an alert, use `sb.accept_alert()`. Otherwise, you either have an incorrect selector (use a valid one), or the button is in an iframe (in which case, switch into the iframe first).

**Reply by davidjivan** [reply]
Thanks! I meant to write that the element is a button, essentially like this: `<button>"I Accept"</button>
`
Which is why I was pretty sure the selector should work:

```
# Constant
ACCEPT_BUTTON = "//button[contains(., 'I Accept')]"

# How I'm telling SB to click it
sb.click(ACCEPT_BUTTON)

```

But I totally didn't realize about switching to the iframe first 😅 That was exactly it.

Appreciate the quick response and the help!
